modAionImpl/src/org/aion/zero/impl/sync/SyncStats.java

```diff
@@ -140,118 +122,67 @@ void update(long _blockNumber) {
+     * with respect to the total number of requests made.
      *
-     * @return a hash map in descending order containing peers with underlying percentage of
-     *     requests made by the node
+     * @return log stream with requests statistical data
      */
-    Map<String, Float> getPercentageOfRequestsToPeers() {
-        if (requestsEnabled) {
-            requestsLock.lock();
-
-            try {
-                Map<String, Float> percentageReq = new LinkedHashMap<>();
-
-                float totalReq = 0f;
-
-                // if there are any values the total will be != 0 after this
-                for (RequestCounter rc : requestsToPeers.values()) {
-                    totalReq += rc.getTotal();
-                }
-
-                // resources are locked so the requestsToPeers map is unchanged
-                // if we enter this loop the totalReq is not equal to 0
-                for (Map.Entry<String, RequestCounter> entry : requestsToPeers.entrySet()) {
-                    percentageReq.put(entry.getKey(), entry.getValue().getTotal() / totalReq);
-                }
-
-                return percentageReq.entrySet().stream()
-                        .sorted(Collections.reverseOrder(Map.Entry.comparingByValue()))
-                        .collect(
-                                Collectors.toMap(
-                                        Map.Entry::getKey,
-                                        Map.Entry::getValue,
-                                        (e1, e2) -> e2,
-                                        LinkedHashMap::new));
-            } finally {
-                requestsLock.unlock();
-            }
+    public String dumpRequestStats() {
+        if (requestEnabled) {
+            return requestTracker.dumpRequestStats();
         } else {
-            return Collections.emptyMap();
+            return "";
         }
     }
 
     /**
-     * Updates the total number of blocks received from each seed peer
+     * Returns a log stream containing a list of peers ordered by the total number of blocks
+     * received from each peer used to determine who is providing the majority of blocks, i.e. top
+     * seeds.
      *
-     * @param nodeId peer node display Id
-     * @param totalBlocks total number of blocks received
+     * @return log stream with peers statistical data on seeds
      */
-    public void updatePeerTotalBlocks(String nodeId, int totalBlocks) {
-        if (seedEnabled) {
-            seedsLock.lock();
-            try {
-                if (blocksByPeer.putIfAbsent(nodeId, totalBlocks) != null) {
-                    blocksByPeer.computeIfPresent(nodeId, (key, value) -> value + totalBlocks);
-                }
-            } finally {
-                seedsLock.unlock();
-            }
+    public String dumpTopSeedsStats() {
+        if (seedsEnabled) {
+            return this.seedsTracker.dumpTopSeedsStats();
```

Comment:
I think we could remove `this.` throughout the class, except for the constructor if there's some name overlap

Reply:
removed